Libraries/WebSocket/WebSocketBase.js

```diff
@@ -33,19 +33,31 @@ class WebSocketBase extends EventTarget {
   readyState: number;
   url: ?string;
 
-  constructor(url: string, protocols: ?any) {
+  /**
+   * @constructor WebSocket
+   * @param {String} address Connection address.
+   * @param {String|Array} protocols WebSocket protocols.
+   * @param {Object} options Additional connection options.
+   */
+  constructor(url: string, protocols: ?any, options: ?any) {
     super();
     this.CONNECTING = 0;
     this.OPEN = 1;
     this.CLOSING = 2;
     this.CLOSED = 3;
 
-    if (!protocols) {
-      protocols = [];
+    if (protocols && !Array.isArray(protocols) && typeof protocols === 'object') {
+      // accept the "options" Object as the 2nd argument
```

Comment:
I don't think we should do this. It complicates the type signature once we try to use more precise types than `any?`. Always specify a value for `protocols` -- much simpler that way.